sktime/utils/_testing/_conditional_fixtures.py

```diff
@@ -0,0 +1,181 @@
+    """
+    fixture_vars = _check_list_of_str(fixture_vars, name="fixture_vars")
+
+    # order fixture_vars according to fixture_sequence if provided
+    if fixture_sequence is not None:
+        fixture_sequence = _check_list_of_str(fixture_sequence, name="fixture_sequence")
+        ordered_fixture_vars = []
+        for fixture_var_name in fixture_sequence:
+            if fixture_var_name in fixture_vars:
+                ordered_fixture_vars += [fixture_var_name]
+        fixture_vars = ordered_fixture_vars
+
+    def get_fixtures(fixture_var, **kwargs):
+        res = generator_dict[fixture_var](test_name, **kwargs)
+        if len(res) == 2:
+            fixture_prod = res[0]
+            fixture_names = res[1]
+        else:
+            fixture_prod = res
+            fixture_names = [str(x) for x in res]
+        return fixture_prod, fixture_names
+
+    fixture_prod = [()]
+    fixture_names = [""]
+
+    # we loop over fixture_vars, incrementally going through conditionals
+    for i, fixture_var in enumerate(fixture_vars):
+        old_fixture_vars = fixture_vars[0:i]
+
+        # then take successive left products
+        new_fixture_prod = []
+        new_fixture_names = []
+
+        for j, fixture in enumerate(fixture_prod):
+            fixture_name = fixture_names[j]
+            if i == 0:
+                kwargs = dict()
+            else:
+                kwargs = dict(zip(old_fixture_vars, fixture))
+
+            new_fixtures, new_fixture_names_r = deepcopy(
+                get_fixtures(fixture_var, **kwargs)
+            )
+            new_fixture_prod += [
+                deepcopy(fixture) + (new_fixture,) for new_fixture in new_fixtures
+            ]
+            new_fixture_names += [f"{fixture_name}-{x}" for x in new_fixture_names_r]
+
+        fixture_prod = new_fixture_prod
+        fixture_names = new_fixture_names
+
+    # due to the concatenation, fixture names all start leading "-" which is removed
+    fixture_names = [x[1:] for x in fixture_names]
+
+    # in pytest convention, variable strings are separated by comma
+    fixture_param_str = ",".join(fixture_vars)
+
+    # we need to remove the tuple bracket from singleton
+    #   in pytest convention, only multiple variables (2 or more) are tuples
+    fixture_prod = [_remove_single(x) for x in fixture_prod]
+
+    return fixture_param_str, fixture_prod, fixture_names
+
+
+def _check_list_of_str(obj, name="obj"):
+    """Check whether obj is a list of str.
+
+    Parameters
+    ----------
+    obj : any object, check whether is list of str
+    name : str, default="obj", name of obj to display in error message
+
+    Returns
+    -------
+    obj, unaltered
+
+    Raises
+    ------
+    TypeError if obj is not list of str
+    """
+    if not isinstance(obj, list) or not np.all(isinstance(x, str) for x in obj):
+        raise TypeError(f"{obj} must be a list of str")
```

Comment:
have not yet reviewed the pr, but happened to spot this typo, so might as well add the suggestion.
```suggestion
        raise TypeError(f"{name} must be a list of str")
```